### PR TITLE
Synchronize SMW with app repo

### PR DESCRIPTION
**Pull request by mszabo-wikia**

### CODE REVIEW

File: extensions/SemanticMediaWiki/load.php [hunk 1]
```diff
@@ -77,11 +77,10 @@ public static function initExtension() {
 		// Wikia change - begin
 		// SUS-1232: allow SemanticMediaWiki config variables to be customized via WikiFactory
 
-		/* @var $smwDefaults Array */
-
-		foreach( $smwDefaults as $variableName => $variableDefaultValue ) {
+		/** @var array $smwDefaults */
+		foreach ( $smwDefaults as $variableName => $variableDefaultValue ) {
```
Comment (WikiaTech) [inline extensions/SemanticMediaWiki/load.php:81]: MAJOR Remove this unused "$smwDefaults" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)

File: extensions/SemanticMediaWiki/DefaultSettings.php [hunk 2]
```diff
@@ -155,7 +155,7 @@
 
 ###
 # If you already have custom namespaces on your site, insert
-#    $smwgNamespaceIndex = ???;
+#    $smwDefaults['smwgNamespaceIndex'] = ???;
```
Comment (WikiaTech) [inline extensions/SemanticMediaWiki/DefaultSettings.php:158]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

File: extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php [hunk 3]
```diff
@@ -74,17 +74,18 @@ public function execute( $par ) {
...
-		#	$refreshjob = Job::factory( $row->job_cmd, $title, $blob, $row->job_id );
-		#} else {
-		#	$refreshjob = null;
-		#}
-		$refreshjob = null; // Wikia change - we do not use job DB table for handling tasks queue (SUS-1305)
+		// $jobQueueLookup = $mwCollaboratorFactory->newJobQueueLookup( $this->getStore()
+		//	->getConnection( 'mw.db' ) );
+		// $row = $jobQueueLookup->selectJobRowFor( 'SMW\RefreshJob' );
+
+		// if ( $row !== false ) { // similar to Job::pop_type, but without deleting the job
```
Comment (WikiaTech) [inline extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php:81]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

File: extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php [hunk 4]
```diff
@@ -74,17 +74,18 @@ public function execute( $par ) {
...
+		//	$title = Title::makeTitleSafe( $row->job_namespace, $row->job_title );
+		//	$blob = (string)$row->job_params !== '' ? unserialize( $row->job_params ) : false;
+		//	$refreshjob = Job::factory( $row->job_cmd, $title, $blob, $row->job_id );
+		//} else {
+		// Wikia change - we do not use job DB table for handling tasks queue (SUS-1305)
+			$refreshjob = null;
+		//}
```
Comment (WikiaTech) [inline extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php:88]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

File: extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php [hunk 5]
```diff
@@ -74,17 +74,18 @@ public function execute( $par ) {
...
 
 		/**** Execute actions if any ****/
 		switch ( $this->getRequest()->getText( 'action' ) ) {
```
Comment (WikiaTech) [inline extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php:91]: MAJOR Add a "case default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)

File: extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php [hunk 6]
```diff
@@ -74,17 +74,18 @@ public function execute( $par ) {
...
-		#if ( $row !== false ) { // similar to Job::pop_type, but without deleting the job
-		#	$title = Title::makeTitleSafe( $row->job_namespace, $row->job_title );
-		#	$blob = (string)$row->job_params !== '' ? unserialize( $row->job_params ) : false;
-		#	$refreshjob = Job::factory( $row->job_cmd, $title, $blob, $row->job_id );
-		#} else {
-		#	$refreshjob = null;
-		#}
-		$refreshjob = null; // Wikia change - we do not use job DB table for handling tasks queue (SUS-1305)
+		// $jobQueueLookup = $mwCollaboratorFactory->newJobQueueLookup( $this->getStore()
+		//	->getConnection( 'mw.db' ) );
```
Comment (WikiaTech) [inline extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php:78]: MAJOR Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 52 issues
* ![MAJOR][MAJOR] 20 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 31 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DefaultSettings.php#L262](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L262): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [DefaultSettings.php#L332](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L332): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [DefaultSettings.php#L370](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L370): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [DefaultSettings.php#L457](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L457): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [DefaultSettings.php#L510](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L510): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [DefaultSettings.php#L512](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L512): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [DefaultSettings.php#L514](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L514): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [DefaultSettings.php#L515](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L515): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [DefaultSettings.php#L516](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L516): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [DefaultSettings.php#L552](https://github.com/Wikia/app/blob/ede1ddddf07ec913a4abe1f8138ed4f6fe596312/extensions/SemanticMediaWiki/DefaultSettings.php#L552): Remove this commented out code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by sqreek — APPROVED** [review]

File: extensions/SemanticMediaWiki/load.php (hunk 1 above)
Comment (mszabo-wikia) [reply extensions/SemanticMediaWiki/load.php:81]: unused? good joke

File: extensions/SemanticMediaWiki/DefaultSettings.php (hunk 2 above)
Comment (mszabo-wikia) [reply extensions/SemanticMediaWiki/DefaultSettings.php:158]: this is documentation

File: extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php (hunk 3 above)
Comment (mszabo-wikia) [reply extensions/SemanticMediaWiki/includes/specials/SMW_SpecialSMWAdmin.php:81]: we need it for future upgrades (but who am I kidding?)

**Pull request merged**